Sources/Commands/SwiftTestTool.swift

```diff
@@ -280,20 +288,51 @@ public class SwiftTestTool: SwiftTool<TestToolOptions> {
                 numJobs: options.numberOfWorkers ?? ProcessInfo.processInfo.activeProcessorCount,
                 diagnostics: diagnostics,
                 options: self.options,
-                buildParameters: try self.buildParameters()
+                buildParameters: buildParameters
             )
             try runner.run(tests)
 
             if !runner.ranSuccessfully {
                 executionStatus = .failure
             }
         }
+
+        if options.shouldEnableCodeCoverage {
+            let buildParameters = try self.buildParameters()
+            let toolChain = try getToolchain()
+            try mergeRawDataFiles(fs: fileManager, path: buildParameters.buildPath.appending(component: "codecov"), toolChain: toolChain)
+            let graph = try loadPackageGraph()
+            for packages in graph.rootPackages {
+                let testPath = try buildTestsIfNeeded(options, graph: graph, binary: true)
+                try generateReport(fs: fileManager, testPath: testPath, buildPath: buildParameters.buildPath, name: packages.name, toolChain: toolChain)
+            }
+        }
+    }
+
+    private func mergeRawDataFiles(fs: FileSystem, path: AbsolutePath, toolChain: UserToolchain) throws {
```

Comment:
doc comment pls

Reply:
I don't think we even need to pass these things as parameters as we're already inside of SwiftTestTool class. We can compute all this inside the method directory. Also, I would suggest calling this method "mergeCodeCovRawDataFiles".

Reply:
done